ddtrace/tracer/tracer.go

```diff
@@ -270,6 +270,7 @@ func newTracer(opts ...StartOption) *tracer {
 		t.reportHealthMetrics(statsInterval)
 	}()
 	t.stats.Start()
+	startTelemetry(c)
```

Comment:
Is there not a check we should do for whether telemetry is enabled first, before we start it by default?

Reply:
There should be logic that checks whether telemetry is enabled when the `GlobalClient` is initialized, as well as when the `GlobalClient` starts. I wanted to move as much telemetry-related code to the telemetry package itself to minimize the code changes to the tracer/profiler. 

But now that I think about it this function name is mis-leading and may be worrying for users who assume telemetry is started no matter what. There is also additional work being inside `startTelemetry` creating config data even if telemetry is disabled. 

I will add a check at the start of `startTelemetry` to immediately return if `telemetry.GlobalClient.Disabled` is true and try to think of some other function name and/or add some comments clarifying when telemetry is actually started.